### PR TITLE
[MOEN-31747] flutter platform core and web-specific changes for object support in user and event attributes

### PR DESCRIPTION
**Pull request by adityagupta-7**
### Jira Ticket
https://moengagetrial.atlassian.net/browse/MOEN-31747
### Description
flutter platform core and web-specific changes for object support in user and event attributes

### CODE REVIEW

**Review by nawazMoEngage — DISMISSED** [review]

File: core/moengage_flutter_platform_interface/lib/src/model/properties.dart [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import 'dart:convert';
+
+import '../../moengage_flutter_platform_interface.dart';
```
Comment (msoumya-engg-sdk) [inline core/moengage_flutter_platform_interface/lib/src/model/properties.dart:3]: Verify if this import needed

**Review by msoumya-engg-sdk — CHANGES_REQUESTED** [review]

File: core/moengage_flutter_web/lib/moengage_flutter_web.dart [hunk 2]
```diff
@@ -1,4 +1,8 @@
+import 'dart:async';
+import 'dart:convert';
 import 'dart:js';
+import 'dart:js_interop';
+import 'package:flutter/foundation.dart';
```
[the remark this answers is not in the document]
Comment (adityagupta-7) [reply core/moengage_flutter_web/lib/moengage_flutter_web.dart]: removed these imports

File: core/moengage_flutter_web/lib/moengage_flutter_web.dart [hunk 3]
```diff
@@ -15,6 +19,11 @@ class MoEngageFlutterWeb extends MoEngageFlutterPlatform {
   @override
   void initialise(MoEInitConfig moEInitConfig, String appId) {
     Logger.d('initialise() : Initialising MoEngage web SDK');
+    Timer(const Duration(seconds: 5), anyFunction);
```
[the remark this answers is not in the document]
Comment (adityagupta-7) [reply core/moengage_flutter_web/lib/moengage_flutter_web.dart]: commented this line and added another comment about its usage

File: core/moengage_flutter_platform_interface/lib/src/model/properties.dart (hunk 1 above)
Comment (adityagupta-7) [reply core/moengage_flutter_platform_interface/lib/src/model/properties.dart]: removed the second import.. first one is required

**Comment by msoumya-engg-sdk** [comment]
@adityagupta-7 please add entry to changelogs for changes packages, rest everything are fine

**Review by msoumya-engg-sdk — DISMISSED** [review]

File: core/moengage_flutter_web/lib/moengage_flutter_web.dart [hunk 4]
```diff
@@ -15,6 +16,14 @@ class MoEngageFlutterWeb extends MoEngageFlutterPlatform {
   @override
   void initialise(MoEInitConfig moEInitConfig, String appId) {
     Logger.d('initialise() : Initialising MoEngage web SDK');
+    // uncomment the below line if moengage web sdk initialisation issue occurs
```
Comment (kkgowtham-engg-sdk) [inline core/moengage_flutter_web/lib/moengage_flutter_web.dart:19]: I think at any time we shouldn't have this generic for all the clients. It should be responsible for the client to call `initialise` at delay based on their load time.

We can just call out in documentation and update Readme.md file [here](https://github.com/moengage/Flutter-SDK/pull/115/files).

File: core/moengage_flutter_web/lib/utils.dart [hunk 5]
```diff
@@ -37,5 +37,8 @@ dynamic getUserAttributeValuePayload(dynamic userAttributeValue) {
   if (userAttributeValue is List<dynamic>) {
     return js.JsArray.from(userAttributeValue);
   }
+  if (userAttributeValue is Map<String, dynamic>) {
+    return js.JsObject.jsify(userAttributeValue as Object);
+  }
```
Comment (kkgowtham-engg-sdk) [inline core/moengage_flutter_web/lib/utils.dart:42]: I think we can remove typecasting to `Object` since all the dart classes have base class as `Object`.

Comment (adityagupta-7) [reply core/moengage_flutter_web/lib/utils.dart:42]: done

File: core/moengage_flutter_web/lib/moengage_flutter_web.dart (hunk 4 above)
Comment (adityagupta-7) [reply core/moengage_flutter_web/lib/moengage_flutter_web.dart]: added this piece of information in README file and removed the comments from the file

**Review by kkgowtham-engg-sdk — DISMISSED** [review]

**Review by kkgowtham-engg-sdk — APPROVED** [review]

**Review by kkgowtham-engg-sdk — DISMISSED** [review]

**Comment by kkgowtham-engg-sdk** [comment]
TODO: Update the Plugin Base version before Release

**Review by umangmoe — APPROVED** [review]

**Pull request merged**